six/two/two.cpp

```diff
@@ -553,9 +551,28 @@ std::string Two::_fetchPythonError()
             if (format_exception != NULL) {
                 PyObject *fmt_exc = PyObject_CallFunctionObjArgs(format_exception, ptype, pvalue, ptraceback, NULL);
                 if (fmt_exc != NULL) {
+                    // Unlikely to happen but PyList_Size might fail, handle error
+                    Py_ssize_t len = PyList_Size(fmt_exc);
+                    if (len == -1) {
```

Comment:
From the docs it's not obvious https://docs.python.org/2.7/c-api/list.html?highlight=pylist_getitem#c.PyList_Size but if you look at the code, it actually raises error (same as `len` in Python) https://github.com/python/cpython/blob/2.7/Objects/listobject.c#L170